### ISSUE TITLE
Use RETURNING PostgreSQL syntax for inserts, instead of Connection::getLastInsertID

### ISSUE THREAD

**Issue by aotd1** [yiisoft/yii2#5042]
I have PostgreSql 9.3 table:
```sql
CREATE EXTENSION "uuid-ossp";
CREATE TABLE Favorites
(
    id UUID PRIMARY KEY DEFAULT uuid_generate_v1mc() NOT NULL,
    user_id INT NOT NULL,
    created_at TIMESTAMP DEFAULT now() NOT NULL,
    ...
);
```
And simple php code:
```php
    $f = new Favorite();
    $f->attributes = ['user_id' => 22599];
    $f->save();
    var_dump($f->attributes);
```
Returns:
```php
array(7) {
  'id' =>  NULL
  'user_id' =>  int(22599)
  'created_at' =>  NULL
  ...
}
```
Table has no sequences, so saved model has empty `id` property (look at [ActiveRecord::insertInternal](https://github.com/yiisoft/yii2/blob/master/framework/db/ActiveRecord.php#L439) logic).

How I can fix that?
Best way as I think - rewrite ActiveRecord::insertInternal (or Command::insert) for PostgreSQL to use insert with returning statement:
```sql
INSERT INTO favorites (user_id) VALUES (22599) RETURNING id;
```
 
PHP: 5.5.15
Yii2:  2.0.0-beta, dev-master (d83d340)

**Comment by EagleMoor** [comment]
It's very need!

**Comment by samdark** [comment]
Do you want to try implementing it and sending a pull request?

**Comment by nineinchnick** [comment]
I do.

**Comment by EagleMoor** [comment]
If we use custom id name like "uuid" and if 32-bit string, RETURNING uuid don't give result.

Fix it we can by 

```php
beforeInsert() {
  $this->uuid = 'samdoasda-asdasmodasda-sadasda'; // generate value
}
```

But if id generate on db, it's not job.

**Comment by nineinchnick** [comment]
@EagleMoor this is a different issue, please create a new issue or search existing ones for something related to that.

**Comment by EagleMoor** [comment]
@nineinchnick why different? 

```sql
id UUID PRIMARY KEY DEFAULT uuid_generate_v1mc() NOT NULL,
```

id generated by function, they can create string pk (UUID type)
It's but, what after insert with no auto increment index, this index not return.

**Comment by psrustik** [comment]
Issue also actual for Oracle database.
```sql
CREATE TABLE mytable (
  ID           NUMBER(10)    NOT NULL,
  DESCRIPTION  VARCHAR2(50)  NOT NULL,
  CONSTRAINT mytable_pk PRIMARY KEY (ID)
);
CREATE SEQUENCE mytable_seq;
CREATE OR REPLACE TRIGGER mytable_tbi
BEFORE INSERT ON mytable 
FOR EACH ROW
BEGIN
 :new.id := mytable_seq.NEXTVAL;
 -- or some logic
 :new.id := calculate_some_value();
END;
```

If dbuser don't has access for database object *sequense* OR primary key calculating by some logic in stored procedures then ActiveRecord insert failed.

Right way to get PK value through operator *returning*.

See db/ActiveRecord.php lines 463:472

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2/pull/8071

**Comment by klimov-paul** [comment]
Resolved by #8071

**Comment by tsv83** [comment]
This patch doesn't cover all cases. There are such thing as views in postgres, and views can't have a primary key. So schema won't return any primary key for the view, and this patch will never add "RETURNING 'colname'" to inserts to the views.

There should be manual way to set primary key (maybe propagate it's value from model's method primaryKey()?) to use it as RETURNING argument.
